tests/std/tests/P2286R8_text_formatting_range_formatter/test.cpp

```diff
@@ -0,0 +1,431 @@
+        using OuterRange = Range::template RebindElement<const Range>;
+
+        assert(fmt(STR("{}"), OuterRange(array{Range(array{0, 1, 2}), Range(array{3, 4, 5}), Range(array{6, 7, 8})}))
+               == STR("[[0, 1, 2], [3, 4, 5], [6, 7, 8]]"));
+        assert(fmt(STR("{::n}"), OuterRange(array{Range(array{0, 1, 2}), Range(array{3, 4, 5}), Range(array{6, 7, 8})}))
+               == STR("[0, 1, 2, 3, 4, 5, 6, 7, 8]"));
+        assert(fmt(STR("{0:_>{2}n:n}"),
+                   OuterRange(array{Range(array{0, 1, 2}), Range(array{3, 4, 5}), Range(array{6, 7, 8})}), 20, 30)
+               == STR("_____0, 1, 2, 3, 4, 5, 6, 7, 8"));
+
+        assert(fmt(STR("{:_<{}:*^{}:=^#{}x}"),
+                   OuterRange(array{Range(array{0, 1, 2}), Range(array{3, 4, 5}), Range(array{6, 7, 8})}), 90, 25, 5)
+               == STR("[**[=0x0=, =0x1=, =0x2=]**, **[=0x3=, =0x4=, =0x5=]**, **[=0x6=, =0x7=, =0x8=]**]_________"));
+    }
+
+    { // Check formatting range of 'FormatNextArg's.
+        using TestRange = Range::template RebindElement<const FormatNextArg>;
+
+        // TRANSITION, GH-4636 (':' is unnecessary)
+        assert(fmt(STR("{:}"), TestRange(array<FormatNextArg, 1>{}), 0) == STR("[0]"));
+        assert(fmt(STR("{:}"), TestRange(array<FormatNextArg, 1>{}), STR("mmm")) == STR("[mmm]"));
+        assert(fmt(STR("{:}"), TestRange(array<FormatNextArg, 1>{}), FormatAsX{}) == STR("[handle: X]"));
+        assert(fmt(STR("{:}"), TestRange(array<FormatNextArg, 2>{}), FormatAsX{}) == STR("[handle: X, handle: X]"));
+    }
+
+    { // Check formatting range of custom type with 'set_debug_format'.
+        using TestRange = Range::template RebindElement<const Debugable>;
+
+        assert(fmt(STR("{}"), TestRange(array<Debugable, 1>{})) == STR("[debug]"));
+        assert(fmt(STR("{:}"), TestRange(array<Debugable, 2>{})) == STR("[debug, debug]"));
+        assert(fmt(STR("{::}"), TestRange(array<Debugable, 3>{})) == STR("[regular, regular, regular]"));
+        assert(fmt(STR("{::D}"), TestRange(array<Debugable, 4>{})) == STR("[debug, debug, debug, debug]"));
+    }
+}
+
+template <class CharT, class Range>
+void check_incorrect_use_of_range_formatter() {
+    ExpectFormatError<CharT> fmt;
+
+    const array vals = {1, 2, 3};
+
+    { // damaged fields
+        fmt(STR("{"), Range(vals));
+        fmt(STR("{}}"), Range(vals));
+        fmt(STR("{{}"), Range(vals));
+        fmt(STR("{:}}"), Range(vals));
+    }
+
+    { // invalid range-format-spec
+        fmt(STR("{:*,20}"), Range(vals));
+        fmt(STR("{:nn}"), Range(vals));
+        fmt(STR("{:x:}"), Range(vals));
+        fmt(STR("{:x}"), Range(vals));
+        fmt(STR("{:ns}"), Range(vals));
+        fmt(STR("{:n?s}"), Range(vals));
+        fmt(STR("{:m}"), Range(vals));
+        fmt(STR("{:s:x}"), Range(vals));
+        fmt(STR("{:?s:x}"), Range(vals));
+        fmt(STR("{:?}"), Range(vals));
+        fmt(STR("{:?z}"), Range(vals));
+        fmt(STR("{::^20}"), Range(vals));

```

Comment:
I think `{::^20}` should be a valid format string, with `^20` being the underlying spec.

Reply:
Good catch, thanks! It looks like `parse` function tries to interpret second `:` as a `range-fill` character.